omas/machine_mappings/d3d.py

```diff
@@ -31,255 +31,320 @@ def gas_injection_hardware(ods, pulse):
     if pulse < 100775:
         warnings.warn('DIII-D Gas valve locations not applicable for pulses earlier than 100775 (2000 JAN 17)')
 
-    i = 0
+    ip = 0
+    iv = 0
 
     def pipe_copy(pipe_in):
-        pipe_out = ods['gas_injection']['pipe'][i]
+        pipe_out = ods['gas_injection']['pipe'][ip]
         for field in ['name', 'exit_position.r', 'exit_position.z', 'exit_position.phi']:
             pipe_out[field] = pipe_in[field]
-        vvv = 0
-        while f'valve.{vvv}.identifier' in pipe_in:
-            valve_identifier = pipe_in[f'valve.{vvv}.identifier']
-            vvv += 1
-        return valve_identifier
+        return
 
     # PFX1
+    valve_pfx1 = ods['gas_injection']['valve'][iv]
+    valve_pfx1['identifier'] = 'PFX1'
+    valve_pfx1['pipe_indices'] = []
     for angle in [12, 139, 259]:  # degrees, DIII-D hardware left handed coords
-        pipe_pfx1 = ods['gas_injection']['pipe'][i]
+        pipe_pfx1 = ods['gas_injection']['pipe'][ip]
         pipe_pfx1['name'] = 'PFX1_{:03d}'.format(angle)
         pipe_pfx1['exit_position']['r'] = 1.286  # m
         pipe_pfx1['exit_position']['z'] = 1.279  # m
         pipe_pfx1['exit_position']['phi'] = -np.pi / 180.0 * angle  # radians, right handed
-        pipe_pfx1['valve'][0]['identifier'] = 'PFX1'
+        pipe_pfx1['valve_indices'] = [iv]
+        valve_pfx1['pipe_indices']=np.append(valve_pfx1['pipe_indices'],[ip])
         dr = -1.116 + 1.286
         dz = -1.38 + 1.279
         # pipea['exit_position']['direction'] = 180/np.pi * tan(dz/dr) if dr != 0 else 90 * sign(dz)
         pipe_pfx1['second_point']['phi'] = pipe_pfx1['exit_position']['phi']
         pipe_pfx1['second_point']['r'] = pipe_pfx1['exit_position']['r'] + dr
         pipe_pfx1['second_point']['z'] = pipe_pfx1['exit_position']['z'] + dz
-        i += 1
+        ip += 1
+    iv += 1
 
     # PFX2 injects at the same poloidal locations as PFX1, but at different toroidal angles
+    valve_pfx2 = ods['gas_injection']['valve'][iv]
+    valve_pfx2['identifier'] = 'PFX2'
+    valve_pfx2['pipe_indices'] = []
     for angle in [79, 199, 319]:  # degrees, DIII-D hardware left handed coords
         pipe_copy(pipe_pfx1)
-        pipe_pfx2 = ods['gas_injection']['pipe'][i]
+        pipe_pfx2 = ods['gas_injection']['pipe'][ip]
         pipe_pfx2['name'] = 'PFX2_{:03d}'.format(angle)
         pipe_pfx2['exit_position']['phi'] = -np.pi / 180.0 * angle  # rad
-        pipe_pfx2['valve'][0]['identifier'] = 'PFX2'
+        pipe_pfx2['valve_indices'] = [iv]
+        valve_pfx2['pipe_indices']=np.append(valve_pfx2['pipe_indices'],[ip])
```

Comment:
You can run `make format` at the top level `omas` folder to format all the files in the OMAS package

Reply:
Thanks for the suggestion!  I forgot that the linter needed to be run by hand.  The formatting has all been fixed now